lib/internal/module.js

```diff
@@ -44,3 +48,31 @@ function stripBOM(content) {
   }
   return content;
 }
+
+exports.builtinLibs = ['assert', 'buffer', 'child_process', 'cluster',
+  'crypto', 'dgram', 'dns', 'domain', 'events', 'fs', 'http', 'https', 'net',
+  'os', 'path', 'punycode', 'querystring', 'readline', 'repl', 'stream',
+  'string_decoder', 'tls', 'tty', 'url', 'util', 'v8', 'vm', 'zlib'];
+
+function addBuiltinLibsToObject(object) {
+  // Make built-in modules available directly (loaded lazily).
+  exports.builtinLibs.forEach((name) => {
+    Object.defineProperty(object, name, {
+      get: () => {
+        const lib = require(name);
+        // This implicitly invokes the setter, so that this getter is only
+        // invoked at most once and does not overwrite anything.
+        object[name] = lib;
+        return lib;
+      },
+      // Allow the creation of other globals with this name.
+      set: (val) => {
+        // Deleting the property before re-assigning it disables the
+        // getter/setter mechanism.
+        delete object[name];
+        object[name] = val;
+      },
+      configurable: true
```

Comment:
It would leave the property non-enumerable in all cases, even when manually overwritten by the user. If that’s okay, sure, that basically works (It has to be `delete object[name]` where you wrote `void object[name]` though, so that the setter does not enter infinite recursion).